tracing/src/main/java/com/palantir/tracing/DetachedSpan.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+/**
+ * Span operation which is not bound to thread state, and can measure operations which
+ * themselves aren't bound to individual threads.
+ */
+public interface DetachedSpan {
+
+    /**
+     * Equivalent to {@link Tracer#startSpan(String)}, but using this {@link DetachedSpan} as the parent instead
+     * of thread state.
+     */
+    SpanToken startSpan(String operationName);
```

Comment:
todo: `startSpan(operation, type)`

Reply:
This API uses a Closeable token rather than `runWith(operation, runnable)` or `callWith(operation, callable)` because it's more versatile in some situations when the operation we wish to trace is not itself a standard java runnable/callable, though we lose some comile-time safety that the token is not passed to another thread.

Reply:
This method name should make it clear that we're setting state on the current thread. `startSpanWithThreadState`?